sys/action.js

```diff
@@ -137,7 +137,14 @@ function buildQueryResponse(res) {
     } else if (!res.body || res.body.error) {
         throw apiError((res.body || {}).error);
     } else if (!res.body.query || (!res.body.query.pages && !res.body.query.userinfo)) {
-        throw apiError({ info: 'Missing query pages from the PHP action API response.' });
+        throw new HTTPError({
+            status: 404,
+            body: {
+                type: 'not_found#page_revisions',
```

Comment:
The question seems to be if there can be other possible reasons for having no query.pages. If so, then we should keep the old behavior, and let apiError handle the overall error. In this particular case we seem to be getting a generic 500 though, and I agree with @Pchelolo that matching the error string is rather hacky.

Could we perhaps pass a status to apiError, or improve apiError to handle this case better?